### PR TITLE
Fix workers AB

### PR DESCRIPTION
**Pull request by midas-myth**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208083051719312

### CODE REVIEW

File: src/lib/multicall/Multicall.ts [hunk 1]
```diff
@@ -172,7 +179,20 @@ export class Multicall {
       sleep(maxTimeout).then(() => Promise.reject(new Error("multicall timeout"))),
     ]).catch((_viemError) => {
       const e = new Error(_viemError.message.slice(0, 150));
-
+      globalThis.dispatchEvent(
+        new CustomEvent("metrics-mark", {
+          detail: {
+            event: "multicall-timeout",
```
Comment (divhead) [inline src/lib/multicall/Multicall.ts:185]: event name differs from types

File: src/lib/multicall/Multicall.ts [hunk 2]
```diff
@@ -172,7 +179,20 @@ export class Multicall {
...
+            isError: true,
+            message: _viemError.message.slice(0, 150),
+            data: {
+              metricType: "multicall.timeout",
+              isInMainThread: !isWebWorker,
+            } satisfies MulticallMetricData,
+            time: Date.now(),
```
Comment (divhead) [inline src/lib/multicall/Multicall.ts:192]: time uses for time diffs, not just “time” 
you can send data.timestamp if needed

File: src/lib/multicall/Multicall.ts (hunk 1 above)
Comment (midas-myth) [reply src/lib/multicall/Multicall.ts]: fixed and typed

File: src/lib/multicall/Multicall.ts (hunk 2 above)
Comment (midas-myth) [reply src/lib/multicall/Multicall.ts]: removed

**Review by divhead — APPROVED** [review]

**Review by divhead — APPROVED** [review]

**Pull request merged**